### PR TITLE
Removed ``Offset``

### LINKED ISSUE #311: Refactor ``Offset``
- [ ] Fix curve degree.
- [ ] Move the methods into the classes.

### PR DESCRIPTION
**Pull request by sonomirco**
### What type of PR is this? (check all applicable)

- [X] ♻️ Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

### Description

This PR removed the ``Offset`` class and moved the offset method into the single curve classes.

### Related Tickets & Documents

This PR closes #311 

### Added tests?

- [ ] 👍 yes
- [X] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📓 docs
- [X] 🙅 no documentation needed

### CODE REVIEW

**Comment by sonomirco** [comment]
@cesarecaoduro @d3ssy in this PR was merged this PR #323

File: src/GShark.Test.XUnit/Operation/ModifyTests.cs [hunk 1]
```diff
@@ -305,23 +303,65 @@ public void Returns_A_Curve_Joining_Polylines_And_Lines()
...
-            Point3 pt1 = joinedCurve.PointAt(0.1);
-            Point3 pt2 = joinedCurve.PointAt(0.25);
-            Point3 pt3 = joinedCurve.PointAt(0.75);
+            NurbsCurve joinedCurve = Modify.JoinCurves(new List<NurbsCurve> { poly.ToNurbs(), ln0.ToNurbs(), ln1.ToNurbs() });
+            Point3 pt1 = joinedCurve.PointAtLength(15);
+            Point3 pt2 = joinedCurve.PointAtLength(21.5);
+            Point3 pt3 = joinedCurve.PointAtLength(27.5);
+            Point3 pt4 = joinedCurve.PointAtLength(35.2);
 
             // Arrange
```
Comment (d3ssy) [inline src/GShark.Test.XUnit/Operation/ModifyTests.cs:322]: Assert

File: src/GShark.Test.XUnit/Operation/ModifyTests.cs [hunk 2]
```diff
@@ -305,23 +303,65 @@ public void Returns_A_Curve_Joining_Polylines_And_Lines()
...
             joinedCurve.Degree.Should().Be(1);
+            (joinedCurve.Length - expectedLength).Should().BeLessThan(GSharkMath.MinTolerance);
             pt1.Equals(expectedPt1).Should().BeTrue();
             pt2.Equals(expectedPt2).Should().BeTrue();
             pt3.Equals(expectedPt3).Should().BeTrue();
+            pt4.Equals(expectedPt4).Should().BeTrue();
```
Comment (d3ssy) [inline src/GShark.Test.XUnit/Operation/ModifyTests.cs:328]: Again here we should just do joinedCurve.Equals(expectedCurve) to test for equality, and expectedCurve defined explicitly in Arrange section.

File: src/GShark/Geometry/Line.cs [hunk 3]
```diff
@@ -197,6 +198,23 @@ public Line Flip()
             return new Line(EndPoint, StartPoint);
         }
 
+        /// <summary>
+        /// Computes the offset of the line.
```
Comment (d3ssy) [inline src/GShark/Geometry/Line.cs:202]: Should specify in which direction the offset is created relative to plane?

**Review by d3ssy — APPROVED** [review]

**Pull request merged**